### PR TITLE
Re-enable doc uploading on Travis

### PR DESCRIPTION
**Pull request by tvolkert**
https://github.com/flutter/flutter/issues/10940

### CODE REVIEW

**Review by goderbauer — APPROVED** [review]
LGTM

File: .travis.yml [hunk 1]
```diff
@@ -1,6 +1,7 @@
 os:
   - linux
 sudo: false
+filter_secrets: false
```
Comment (Hixie) [inline .travis.yml]: why?

Comment (tvolkert) [reply .travis.yml]: Was trying to fix #10940, and in an email thread with Travis support, I received the following:

> I've been reviewing the possible differences with one of my teammates and we believe that it may be caused by some interaction of the filter we apply to push builds to avoid accidentally outputting secure data in the build log, see: https://docs.travis-ci.com/user/best-practices-security/#Steps-Travis-CI-takes-to-secure-your-data
>
> Although we're not 100% sure this would fix it, would you be open to switching the script from curl -# -f https://dl.google.com/dl/cloudsdk/channels/rapid/google-cloud-sdk.tar.gz to 
>
> curl -f https://dl.google.com/dl/cloudsdk/channels/rapid/google-cloud-sdk.tar.gz
>
> removing the progress bar from it? 
>
> Another option, to rule in/out the filter as a possible culprit would be deactivating it entirely, by adding the following snippet in the top level of .travis.yml:
>
> filter_secrets: false
>
> This will disable the extra-filtering of the contents of environment variables marked as secret in the build log, should they accidentally be leaked there, and will bring you back to the situation as it was before May.

The `curl` trick didn't work, and the `filter_secrets` trick did.

TLDR: it's a workaround to a probably bug on Travis' side of things.
